### DISCUSSION TITLE
THB2 - new (incompatible) hardware revision

### DISCUSSION THREAD

**Discussion by greedysquid** [pvvx/THB2#92] [General]
Firstly, thank you for the developing and sharing this f/w!

Have been using this repo to create BTHome sensors to use with my Home Assistant. 
Started with a cheap BTH01 and TH05F, but then realised these a poor choice because of the VCC > 2.5V requirement of CHT8305 sensor.
Therefore recently ordered some THB2 (which should have a CHT8310 sensor), from: https://www.aliexpress.com/item/1005007987282700.html

Posting this because it seems there has been an iteration of the THB2 hardware.
The THB2 that arrived have the CHT8305 sensor (bad!) and a 'BTU' Bluetooth module.
I _think_ the BTU module is: https://developer.tuya.com/en/docs/iot/btu-module-datasheet?id=Kakc18n7eo9k3
The module spec says is based on a different BT device (TLSR8250), and requires VCC 2.8V - 3.6V.

If this is correct, then several implications:
- this firmware can't be used with this (new?) version of THB2
- even if a compatible firmware, VCC > 2.8V means this hardware even more unsuitable for operation from 2x AA
- not sure how to identify this (new?) version of hardware when ordering

☹️

![IMG_20241204_111304141](https://github.com/user-attachments/assets/82490f8a-fc41-492f-98f3-fa9563324607)
![IMG_20241204_111346632](https://github.com/user-attachments/assets/e87f9e68-80bc-4b1f-a097-a40f41a8a3b5)
![IMG_20241204_111427198](https://github.com/user-attachments/assets/d51bc1f4-35cd-4dce-97b4-ff72d79c9a35)
![IMG_20241204_113448896](https://github.com/user-attachments/assets/930316bc-fa75-4152-bdd5-c3381e877411)

**Comment by pvvx** [comment]
TLSR8250 -  Power supply: 1.8..3.6V (0FN32 package) / 2.7..3.6V (TSSOP16 package).
[TS0201](https://pvvx.github.io/TS0201_TZ3000)
https://github.com/pvvx/ATC_MiThermometer

**Comment by greedysquid** [comment]
Thank you @pvvx ! 
For taking the time to reply, and pointing me towards your alternative BLE firmware for TLSR825x devices.
And apologies for the slow follow up - has taken me a while to get back to this and work though your suggestions.

However, I have been able to connect to this BTU/TLSR8250 based THB2 sensor using the [`TLSR825xComFlasher.py`](https://github.com/pvvx/TlsrComSwireWriter) utility and successfully flash the `TS0201_v48.bin` image.

Question: what are the differences between `TS0201_v48.bin`, `TS0201S1_v48.bin` and `TS0201S2_v48.bin` firmwares?

Also found some extra info about the supply voltage range from the Tuya BTU module [datasheet](https://developer.tuya.com/en/docs/iot/btu-module-datasheet?id=Kakc18n7eo9k3#title-1-Features):
>Working voltage: 1.8 to 3.6 V. If the voltage is between 1.8 and 2.7 V, the module can start, but the optimal RF performance cannot be ensured. If the voltage is between 2.8 and 3.6 V, the overall performance is normal.

Unclear if the BTU module uses the QFN32 or TSSOP16 package. Probably the former from this [pic](https://developer.tuya.com/en/docs/iot/btu-module-datasheet?id=Kakc18n7eo9k3#title-5-Pin%20definition).
Either way, this module is a poor design choice by Tuya to be powered from 2x AA, but the BT should at least continue to operate (to some level) as the battery discharges.

End result: I do now have some working new-THB2 sensors broadcasting BTHome data and being received by Home Assistant.
Thanks again!

**Comment by pvvx** [comment]
> Question: what are the differences between TS0201_v48.bin, TS0201S1_v48.bin and TS0201S2_v48.bin firmwares?

https://github.com/pvvx/ATC_MiThermometer/issues/571

TLSR825x (module BTU/ZTU) Working voltage: 1.8..3.6V.
The minimum supply voltage recommended by the manufacturer (Telink) for TLSR825x is 2.2 V. This is for the case of active Flash writing, taking into account that during operation, the battery voltage may drop due to the load current. The actual minimum operating voltage is 1.8V.

https://github.com/pvvx/ZigbeeTLc/issues/37#issuecomment-2543039863

CHT8305 Working voltage: min 2.5V - At lower supply voltage, temperature and humidity readings begin to drop sharply, by tens of degrees from the actual ones.

**Reply by greedysquid** [reply]
Thanks for the extra info and explanation on firmware variations - looks interesting, may have to play with adding external sensor(s)...
